build/source/engine/allocspace.f90

```diff
@@ -19,405 +19,348 @@
 ! along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 module allocspace_module
+! data types
 USE nrtype
+! metadata structures
+USE globalData,only:time_meta,forc_meta,attr_meta,type_meta        ! metadata structures
+USE globalData,only:prog_meta,diag_meta,flux_meta,deriv_meta       ! metadata structures
+USE globalData,only:mpar_meta,indx_meta                            ! metadata structures
+USE globalData,only:bpar_meta,bvar_meta                            ! metadata structures
+USE globalData,only:model_decisions                                ! model decision structure
+! time structures
+USE globalData,only:refTime,startTime,finshTime                    ! reference time, start time, and end time for the model simulation
+! structure information
+USE globalData,only:structInfo             ! information on the data structures                  
+! provide access to the derived types to define the data structures
+USE data_types,only:&
+                    var_i,               & ! x%var(:)            (i4b)
+                    var_d,               & ! x%var(:)            (dp)
+                    var_ilength,         & ! x%var(:)%dat        (i4b)
+                    var_dlength,         & ! x%var(:)%dat        (dp)
+                    spatial_int,         & ! x%hru(:)%var(:)     (i4b)
+                    spatial_double,      & ! x%hru(:)%var(:)     (dp)
+                    spatial_intVec,      & ! x%hru(:)%var(:)%dat (i4b)
+                    spatial_doubleVec      ! x%hru(:)%var(:)%dat (dp)
+! metadata structure
+USE data_types,only:var_info
 implicit none
 private
-public::init_metad
-public::alloc_stim
-public::alloc_time
-public::alloc_forc
-public::alloc_attr
-public::alloc_type
-public::alloc_mpar
-public::alloc_mvar
-public::alloc_indx
-public::alloc_bpar
-public::alloc_bvar
+public::initStruct
+public::allocLocal
 ! define missing values
 integer(i4b),parameter :: missingInteger=-9999
 real(dp),parameter     :: missingDouble=-9999._dp
```

Comment:
Elsewhere `-999` is used (in var_lookup.f90 - which is a different use case). Just wondering whether it would be useful to be consistent. Also note that we could use one of the netcdf `_FillValue` values for this.